clippy_lints/src/redundant_pattern_matching.rs

```diff
@@ -40,10 +44,11 @@ use crate::rustc_errors::Applicability;
 /// if Err::<i32, i32>(42).is_err() {}
 /// if None::<()>.is_none() {}
 /// if Some(42).is_some() {}
+/// Ok::<i32, i32>(42).is_ok();
 /// ```
 ///
 declare_clippy_lint! {
-    pub IF_LET_REDUNDANT_PATTERN_MATCHING,
+    pub REDUNDANT_PATTERN_MATCHING,
```

Comment:
In order to properly deprecate the lint, you will have to add the old one to [deprecated_lints.rs](https://github.com/rust-lang-nursery/rust-clippy/blob/master/clippy_lints/src/deprecated_lints.rs)

Reply:
Apart from that, the PR looks good to me :+1:

Reply:
Ah, didn't notice that! Thanks for catching this! I'll update it later!